File: library/src/main/java/io/github/anderscheow/validator/rules/common/ContainRule.java
package io.github.anderscheow.validator.rules.common;

import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import java.util.Locale;

import io.github.anderscheow.validator.rules.BaseRule;

public class ContainRule extends BaseRule {

    private String keyword;

    public ContainRule(String keyword) {
        this.keyword = keyword;
    }

    public ContainRule(String keyword, @StringRes int errorRes) {
        super(errorRes);
        this.keyword = keyword;
    }

    public ContainRule(String keyword, @NonNull String errorMessage) {
        super(errorMessage);
        this.keyword = keyword;
    }

    @Override
    public boolean validate(String value) {
        return value.contains(keyword);
    }

    @NonNull
    @Override
    public String errorMessage() {
        return String.format(Locale.getDefault(), "Value does not contain '%s'", keyword);
    }
}

File: library/src/main/java/io/github/anderscheow/validator/rules/common/ContainRule.kt
package io.github.anderscheow.validator.rules.common

import androidx.annotation.StringRes
import io.github.anderscheow.validator.rules.Rule

class ContainRule : Rule {

    private var keyword: String

    private var ignoreCase = false

    constructor(keyword: String, @StringRes errorRes: Int, ignoreCase: Boolean = false) :
            super(errorRes) {
        this.keyword = keyword
        this.ignoreCase = ignoreCase
    }

    constructor(keyword: String, errorMessage: String, ignoreCase: Boolean = false) :
            super(errorMessage) {
        this.keyword = keyword
        this.ignoreCase = ignoreCase
    }

    override fun validate(value: String?): Boolean {
        if (value == null) {
            throw NullPointerException()
        } else {
            return value.contains(keyword, ignoreCase)
        }
    }
}

fun contain(
    keyword: String,
    @StringRes errorRes: Int,
    ignoreCase: Boolean = false
): ContainRule = ContainRule(keyword, errorRes, ignoreCase)

fun contain(
    keyword: String,
    errorMessage: String,
    ignoreCase: Boolean = false
): ContainRule = ContainRule(keyword, errorMessage, ignoreCase)